source/file_utils.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/io/file_utils.h>
+
+#include <errno.h>
+#include <stdio.h>
+
+int aws_byte_buf_init_from_file(struct aws_byte_buf *out_buf, struct aws_allocator *alloc, const char *filename) {
+#ifdef _MSC_VER
```

Comment:
Could we just do this at file scope? It seems weird to do it inside a function

Reply:
Sure, will update

Reply:
As far as channel vs. other names goes, I know everyone would prefer to not have an overlap with the existing aws_channel, but I feel vocabulary-wise, channel is a much more accurate description (that is in sync with its recent usage within programming languages ala Rust/Go) than anything else proposed.  Pipeline feels like a mismatch given that's how we're describing the sequence of components that make up the default logger.